stream_alert/shared/metrics.py

```diff
@@ -13,14 +13,25 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
-from datetime import datetime
 import json
 import os
 
-import boto3
-from botocore.exceptions import ClientError
-
 from stream_alert.shared import LOGGER
+# Since this is a shared package, try to import the relevant processor name
+try:
+    from stream_alert.alert_processor import FUNCTION_NAME as ALERT_PROCESSOR
+except ImportError:
+    ALERT_PROCESSOR = 'alert_processor'
+
+try:
+    from stream_alert.athena_partition_refresh import FUNCTION_NAME as ATHENA_PROCESSOR
+except ImportError:
+    ATHENA_PROCESSOR = 'athena_partition_refresh'
+
+try:
+    from stream_alert.rule_processor import FUNCTION_NAME as RULE_PROCESSOR
+except ImportError:
+    RULE_PROCESSOR = 'rule_processor'
```

Comment:
I don't really like these `try: except` imports with default overrides. This could lead to an inconsistency where the`MetricLogger._available_metrics` dict has different keys in each Lambda function if the default values here differ from the `FUNCTION_NAME`. That might not matter, but it makes it more difficult to reason about.

The `shared` code is uploaded with every Lambda function, right? Maybe we could define all function names in the shared module?

Reply:
That's not a bad suggestion - let me look into placing in `shared`. I was super confident with this approach anyway